src/ArborX_DBSCAN.hpp

```diff
@@ -51,25 +54,25 @@ template <typename Primitives>
 struct PrimitivesWithRadius
 {
   Primitives _primitives;
-  double _r;
+  float _r;
 };
 
 template <typename Primitives, typename PermuteFilter>
 struct PrimitivesWithRadiusReorderedAndFiltered
 {
   Primitives _primitives;
-  double _r;
+  float _r;
   PermuteFilter _filter;
 };
 
 // Mixed primitives consist of a set of boxes corresponding to dense cells,
 // followed by boxes corresponding to points in non-dense cells.
-template <typename PointPrimitives, typename DenseCellOffsets,
+template <typename PointPrimitives, typename Grid, typename DenseCellOffsets,
```

Comment:
I really dislike the amount/mix of template parameters here.
You can deduce the dimension and then the grid type from `PointPrimitives below`

Reply:
Done. It's ugly, but will be fixed with #729.